None of the STLport versions I have looked at include the version number
in their name, static or not.

--- boost_1_44_0.orig/tools/build/v2/tools/stlport.jam	2009-05-22 21:36:46.000000000 +1000
+++ boost_1_44_0/tools/build/v2/tools/stlport.jam	2010-10-04 18:22:02.592121895 +1000
@@ -142,14 +142,6 @@
                 name += _static ;
             }
 
-            # Starting with version 5.2.0, the STLport static libraries no longer
-            #  include a version number in their name
-            local version.pre.5.2 = [ MATCH "^(5[.][01]+).*" : $(version) ] ;
-            if $(version.pre.5.2) || [ feature.get-values <runtime-link> : $(raw) ] != "static" 
-            {
-                name += .$(self.version.5) ;
-            }
-			
             name = $(name:J=) ;
             
             if [ feature.get-values <install-dependencies> : $(raw) ] = "on"
